umi_tools/extract.py

```diff
@@ -285,7 +285,13 @@ def main(argv=None):
                               add_sam_options=False)
 
     if options.retain_umi and not options.extract_method == "regex":
-        raise ValueError("option --retain-umi only works with --extract-method=regex")
+        U.error("option --retain-umi only works with --extract-method=regex")
+
+    if (options.filtered_out and not options.extract_method == "regex" and
```

Comment:
Should this not be `options.filtered_out and (not options.extract_method == "regex" or not filter_cell_ barcodes)`

Reply:
Ah, no.  `options.filtered_out` is only compatible with either extract using regex (`options.extract_method == "regex" ) *or* cell barcode filtering (`not filter_cell_ barcodes`), since these are the only times one wants to filter the reads in any way. Hence, error is raise if both are false.

I guess the simplest to write this would actually be `options.filtered_out and not (options.extract_method == "regex" or filter_cell_ barcodes)`